packages/liveblocks-client/src/EventQueue.ts

```diff
@@ -0,0 +1,37 @@
+type Callback<T> = (event: T) => void;
+type UnsubscribeCallback = () => void;
+
+/**
+ * makeEventPair allows you to generate a subscribe/notify pair of functions to
+ * make subscribing easy and to get notified about events.
+ *
+ * The events are anonymous, so you can use it to define events, like so:
+ *
+ *   const [subscribeToEvent1, fireEvent1] = makeEventPair();
+ *   const [subscribeToEvent2, fireEvent2] = makeEventPair();
+ *
+ *   subscribeToEvent1(foo);
+ *   subscribeToEvent1(bar);
+ *   subscribeToEvent2(qux);
+ *
+ *   fireEvent1();  // Now foo and bar will get called
+ *   fireEvent2();  // Now qux will get called
+ *
+ */
+export function makeEventPair<T>(): [
+  (callback: Callback<T>) => UnsubscribeCallback,
+  (event: T) => void
+] {
+  const _observers = new Set<Callback<T>>();
+
+  function subscribe(callback: Callback<T>): UnsubscribeCallback {
+    _observers.add(callback);
+    return () => _observers.delete(callback);
+  }
+
+  function notify(event: T) {
+    _observers.forEach((callback) => callback(event));
+  }
+
+  return [subscribe, notify];
+}
```

Comment:
This is a simple helper to encapsulate subscription/notification updates. By creating an event pair this way, all the code to manage those event callbacks get abstracted away, leading to a lot less code (and supporting variables) in the Room. In addition, the split between subscribe and notify means you can publicly expose the `subscribe` part to the world, in an outer API, to allow external subscribers, while keeping the `notify` part for internal-to-the-room use only (so external parties cannot call `notify` themselves).

I would love to restructure all the internal `subscribe()`-related support code in the Room using this abstraction too, but it would be beyond the scope of this PR (and I wouldn't get this done before my holiday anyway).

I'll make a ticket to generalize this implementation clean up the Room internals using this little helper abstraction.